src/main/scala/ai/privado/exporter/CollectionExporter.scala

```diff
@@ -101,6 +107,29 @@ class CollectionExporter(cpg: Cpg) {
         })
     })
 
+    collectionMethods.foreach(collectionMethod => {
+      collectionMethod.literal
+        .and(_.tag.nameExact(Constants.id))
+        .foreach(literal => {
+          try {
+            def addToMap(literalId: String): Unit = {
```

Comment:
@hiteshbedre I see the function `addToMap` getting repeated 3 times, better to make a generic one and use

Reply:
Handled here: https://github.com/Privado-Inc/privado-core/pull/379/commits/49e4553a24590a35a56d5d319be1b0bd81f84b99